### PR TITLE
[fix] sum_to_size: MathBits test - don't reuse same input tensor

### LINKED ISSUE #85409: Result of backward is wrong with conj_view and neg_view for Tensor.sum_to_size
### 🐛 Describe the bug

Newly added sample inputs in https://github.com/pytorch/pytorch/pull/85338 revealed a problem. All I changed was to pass `shape` argument as a list and not a tuple.
```py
_________________________________________________ TestMathBitsCPU.test_conj_view_sum_to_size_cpu_complex64 _________________________________________________
Traceback (most recent call last):
  File "/home/iyashchuk/dev/pytorch/master/test/test_ops.py", line 1454, in test_conj_view
    self._test_math_view(
  File "/home/iyashchuk/dev/pytorch/master/test/test_ops.py", line 1429, in _test_math_view
    self.assertEqual(tensor.grad, cloned1_tensor.grad)
  File "/home/iyashchuk/dev/pytorch/master/torch/testing/_internal/common_utils.py", line 2435, in assertEqual
    assert_equal(
  File "/home/iyashchuk/dev/pytorch/master/torch/testing/_comparison.py", line 1093, in assert_equal
    raise error_metas[0].to_error(msg)
AssertionError: Tensor-likes are not close!

Mismatched elements: 5 / 5 (100.0%)
Greatest absolute difference: 1.551922600020548 at index (0,) (up to 1e-05 allowed)
Greatest relative difference: 1.551922600020548 at index (0,) (up to 1.3e-06 allowed)
__________________________________________________ TestMathBitsCPU.test_neg_view_sum_to_size_cpu_float64 ___________________________________________________
Traceback (most recent call last):
  File "/home/iyashchuk/dev/pytorch/master/test/test_ops.py", line 1473, in test_neg_view
    self._test_math_view(
  File "/home/iyashchuk/dev/pytorch/master/test/test_ops.py", line 1429, in _test_math_view
    self.assertEqual(tensor.grad, cloned1_tensor.grad)
  File "/home/iyashchuk/dev/pytorch/master/torch/testing/_internal/common_utils.py", line 2435, in assertEqual
    assert_equal(
  File "/home/iyashchuk/dev/pytorch/master/torch/testing/_comparison.py", line 1093, in assert_equal
    raise error_metas[0].to_error(msg)
AssertionError: Tensor-likes are not close!

Mismatched elements: 5 / 5 (100.0%)
Greatest absolute difference: 0.5322531452148951 at index (0,) (up to 1e-07 allowed)
Greatest relative difference: 0.5322531452148951 at index (0,) (up to 1e-07 allowed)
________________________________________________ TestMathBitsCUDA.test_conj_view_sum_to_size_cuda_complex64 ________________________________________________
Traceback (most recent call last):
  File "/home/iyashchuk/dev/pytorch/master/test/test_ops.py", line 1454, in test_conj_view
    self._test_math_view(
  File "/home/iyashchuk/dev/pytorch/master/test/test_ops.py", line 1429, in _test_math_view
    self.assertEqual(tensor.grad, cloned1_tensor.grad)
  File "/home/iyashchuk/dev/pytorch/master/torch/testing/_internal/common_utils.py", line 2435, in assertEqual
    assert_equal(
  File "/home/iyashchuk/dev/pytorch/master/torch/testing/_comparison.py", line 1093, in assert_equal
    raise error_metas[0].to_error(msg)
AssertionError: Tensor-likes are not close!

Mismatched elements: 5 / 5 (100.0%)
Greatest absolute difference: 0.7916027468331712 at index (0,) (up to 1e-05 allowed)
Greatest relative difference: 0.7916027468331712 at index (0,) (up to 1.3e-06 allowed)
_________________________________________________ TestMathBitsCUDA.test_neg_view_sum_to_size_cuda_float64 __________________________________________________
Traceback (most recent call last):
  File "/home/iyashchuk/dev/pytorch/master/test/test_ops.py", line 1473, in test_neg_view
    self._test_math_view(
  File "/home/iyashchuk/dev/pytorch/master/test/test_ops.py", line 1429, in _test_math_view
    self.assertEqual(tensor.grad, cloned1_tensor.grad)
  File "/home/iyashchuk/dev/pytorch/master/torch/testing/_internal/common_utils.py", line 2435, in assertEqual
    assert_equal(
  File "/home/iyashchuk/dev/pytorch/master/torch/testing/_comparison.py", line 1093, in assert_equal
    raise error_metas[0].to_error(msg)
AssertionError: Tensor-likes are not close!

Mismatched elements: 5 / 5 (100.0%)
Greatest absolute difference: 1.0829307052989563 at index (0,) (up to 1e-07 allowed)
Greatest relative difference: 1.0829307052989563 at index (0,) (up to 1e-07 allowed)
================================================================= short test summary info ==================================================================
FAILED test/test_ops.py::TestMathBitsCPU::test_conj_view_sum_to_size_cpu_complex64 - AssertionError: Tensor-likes are not close!
FAILED test/test_ops.py::TestMathBitsCPU::test_neg_view_sum_to_size_cpu_float64 - AssertionError: Tensor-likes are not close!
FAILED test/test_ops.py::TestMathBitsCUDA::test_conj_view_sum_to_size_cuda_complex64 - AssertionError: Tensor-likes are not close!
FAILED test/test_ops.py::TestMathBitsCUDA::test_neg_view_sum_to_size_cuda_float64 - AssertionError: Tensor-likes are not close!
```

### Versions

Master.

cc @ezyang @gchanan @zou3519 @albanD @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7 @anjali411 @dylanbespalko @mruberry

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes #85409

### CODE REVIEW

**Review by anjali411 — APPROVED** [review]
good catch!

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -6246,12 +6246,11 @@ def sample_inputs_sum_to_size(op_info, device, dtype, requires_grad, **kwargs):
...
     for input_shape, output_shape in sample_shapes:
-        input_t = make_arg(input_shape)
-        yield SampleInput(input_t, args=(output_shape,))
+        yield SampleInput(make_arg(input_shape), args=(output_shape,))
         if output_shape == ():
             continue
-        yield SampleInput(input_t, args=(list(output_shape),))
-        yield SampleInput(input_t, args=(*output_shape,))
+        yield SampleInput(make_arg(input_shape), args=(list(output_shape),))
+        yield SampleInput(make_arg(input_shape), args=(*output_shape,))
```
Comment (lezcano) [inline torch/testing/_internal/common_methods_invocations.py:6253]: Don't we have a test to catch this one? I thought that the `contiguous` test should have caught this one. @mruberry

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:6253]: You mean 

https://github.com/pytorch/pytorch/blob/4bfb7341819b3bfcaf65ddc136f25d23983740a7/test/test_ops.py#L443

?

Comment (lezcano) [reply torch/testing/_internal/common_methods_invocations.py:6253]: So, it's not clear to Mike or me why the following assert
https://github.com/pytorch/pytorch/blob/4bfb7341819b3bfcaf65ddc136f25d23983740a7/test/test_ops.py#L461-L462
didn't fire in this case.
That assert detects (in a somewhat indirect way) whether there's two sample inputs that share the same tensor. Could you have a look @kshitij12345?

Comment (kshitij12345) [reply torch/testing/_internal/common_methods_invocations.py:6253]: We don't really call backward on the output (instead we use `torch.autograd.grad`) and hence we never set `.grad` on the input.

https://github.com/pytorch/pytorch/blob/409efebab8718a7bfc714ab3787e5a8689289697/test/test_ops.py#L519-L526

```python
import torch

t = torch.randn(1, requires_grad=True)
out = torch.sum(t)
grads = torch.autograd.grad(out, t)

print(t.grad) # None
out.backward()
print(t.grad) # tensor([1.])
```

Comment (lezcano) [reply torch/testing/_internal/common_methods_invocations.py:6253]: oof. I was the one that wrote that bit of code. I'll send a fix if I find the time tomorrow :)

**Pull request closed without merging**